gpgme-sharp/Context.cs

```diff
@@ -94,51 +85,48 @@ private void Initialize(ref ContextSigners signers, ref ContextSignatureNotation
         }
 
         public void Dispose() {
-            if (_ctx_ptr != IntPtr.Zero) {
-                libgpgme.NativeMethods.gpgme_release(_ctx_ptr);
-                _ctx_ptr = IntPtr.Zero;
+            if (CtxPtr != IntPtr.Zero) {
+                libgpgme.NativeMethods.gpgme_release(CtxPtr);
+                CtxPtr = IntPtr.Zero;
             }
         }
 
-        public bool IsValid {
-            get { return (_ctx_ptr != IntPtr.Zero); }
-        }
-        internal IntPtr CtxPtr {
-            get { return _ctx_ptr; }
-        }
-        internal object CtxLock {
-            get { return _ctx_lock; }
+        public bool IsValid => (CtxPtr != IntPtr.Zero);
+        internal IntPtr CtxPtr { get; private set; } = IntPtr.Zero;
+		internal object CtxLock { get; } = new object();
+
+        public void EnsureValid()

```

Comment:
How can we be sure we won't get race conditions as this method is called just before calling native methods? Previously the check was made in a condition, so when the context was invalidated, the calls to the native methods weren't performed.

Reply:
What's a race condition that could occur with the new code that couldn't occur with the old code? To me it seems like it should be the same behaviour.

Reply:
Yes, indeed nothing new under the tree compared to the old code version.

Reply:
Should I move the `EnsureValid` calls so that they're inside the lock? Not sure if that'd somehow risk hitting a deadlock though.

Reply:
It would be better IMHO. Feel free to open a new PR for this :)